misc/server/src/main.rs

```diff
@@ -114,10 +116,35 @@ async fn main() -> Result<(), Box<dyn Error>> {
     );
     let mut swarm = SwarmBuilder::with_tokio_executor(transport, behaviour, local_peer_id).build();
 
-    if config.addresses.swarm.is_empty() {
-        warn!("No listen addresses configured.");
-    }
-    for address in &config.addresses.swarm {
+    let listen_addresses = {
+        let addresses = config.addresses.swarm.clone();
+        if addresses.is_empty() {
+            warn!("No listen addresses configured.");
+        }
+        // Configuration files generated by Kubo <= v0.22 list both `/quic` and `/quic-v1` listen
```

Comment:
Since `/quic` is deprecated, IMO it be more correct to just filter out any `/quic` records and only add `/quic-v1` if it's explicitly stated in the config. Just replacing `/quic` by `/quic-v1` feels a bit close to magic.

It would also be valid to error out with an appropriate error message if `/quic` is added, although that would break any existing installations :).

How did Kubo do it?

OTOH it's really a matter of opinion. If you think it's good this way let's leave it.

Reply:
> Just replacing /quic by /quic-v1 feels a bit close to magic.

Note the subsequent `sort` and `dedup` calls. Thus we don't just replace `/quic` with `/quic-v1`, but remove thereafter identical addresses. Does that make sense?

Reply:
Note that none of this matters for incoming connections. We accept `/quic` and `/quic-v1` either way.

Reply:
the main issue with the current approach is if you have `quic` but not `quic-v1`, then the net effect would be to replace it.

Reply:
Oh I can't seem to approve it.

Reply:
I agree with @mcamou here. It would be more correct to also support the case where the user wants to only listen on `/quic`.

That is a bit more complicated though because we'd have to group by port and protocol. For how long do we need to support `/quic`? Can we just disable `draft-29` support instead and log a warning that we couldn't listen on this address?

Reply:
I don't think it really matters since `/quic` (i.e. draft-29) is less and less relevant.

> Can we just disable `draft-29` support instead and log a warning that we couldn't listen on this address?

Sounds good to me.

I am not online much today. @thomaseizinger in case you can do that change and merge, great. If not, I can tackle this early next week.